src/items/type-aliases.md

```diff
@@ -20,13 +20,18 @@ type Point = (u8, u8);
 let p: Point = (41, 68);
 ```
 
-A type alias to an enum type cannot be used to qualify the constructors:
+A type alias to an tuple-struct type cannot be used to qualify the constructors:
 
 ```rust
-enum E { A }
-type F = E;
-let _: F = E::A;  // OK
-// let _: F = F::A;  // Doesn't work
+pub struct MyStruct(u32);
+
+pub use self::MyStruct as PubUse;
+pub type PubType = MyStruct;
+
+fn main() {
+    let _ = PubUse(5); // OK
+    // let _ = PubType(5); // Doesn't work
+}
```

Comment:
I think it is good when demonstrating something that doesn't work to actually include the broken code, and mark it as `compile_fail`.  I would also avoid the `main` function and the `pub` qualifications, since they aren't really relevant here.  Maybe something like this:

    ```rust,edition2018,compile_fail
    struct MyStruct(u32);
    use MyStruct as PubUse;
    type PubType = MyStruct;

    let _ = PubUse(5); // OK
    let _ = PubType(5); // ERROR: found type alias `PubType`
    ```

Reply:
I think that the `pub` usage in this case is actually slightly relevant because this is most relevant when dealing with renaming types without making a breaking change to your API, so the type being exported is the primary reason this information is important.

Reply:
I'm not sure I follow.  Type aliases are used in many circumstances unrelated to API compatibility.  I see the primary reason for this example is to demonstrate how tuple struct constructors are not aliased with the `type` item, not to illustrate various use cases for aliases.